### PR TITLE
Remove description from Glance CRD

### PR DESCRIPTION
**Pull request by fmount**
This patch rebuilds the Glance `CRD` removing the description to avoid a potential issue when the Glance bundle is deployed via `OLM` by the meta operator and the following is seen:

```
openstack/install-7ws9s"} failed: failed to update installplan bundle lookups: etcdserver: request is too large
```

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/openstack-k8s-operators/glance-operator
-  newTag: v0.0.1
+  newName: quay.io/openstack-k8s-operators/glance-operator-index
```
Comment (abays) [inline config/manager/kustomization.yaml:15]: This should actually point to the operator image itself, not the index.  But the tag change is correct.

Comment (fmount) [reply config/manager/kustomization.yaml]: sorry my bad on that one, removing the `-index` while bumping the tag

**Comment by fmount** [comment]
@abays @stuggi I'm wondering if  we want to enfore `CRDDESC=0` even in `make manifests`

**Comment by abays** [comment]
> @abays @stuggi I'm wondering if we want to enforce `CRDDESC=0` even in `make manifests`

I'm guessing that removing these descriptions from the bundle and potentially even the CRD manifests will mostly only be noticeable when trying to create the CRs through the OpenShift web console.  The fields there would no longer have the descriptions, but I don't think we're supporting GUI creation of resources anyhow.  Of course, the `oc get crd <glance CRD>` would not have them either.  But I think we need to do this to avoid the problem mentioned in the PR description.

**Comment by fmount** [comment]
> > @abays @stuggi I'm wondering if we want to enforce `CRDDESC=0` even in `make manifests`
> 
> I'm guessing that removing these descriptions from the bundle and potentially even the CRD manifests will mostly only be noticeable when trying to create the CRs through the OpenShift web console. The fields there would no longer have the descriptions, but I don't think we're supporting GUI creation of resources anyhow. Of course, the `oc get crd <glance CRD>` would not have them either. But I think we need to do this to avoid the problem mentioned in the PR description.

Yeah, I agree. I would reduce the scope to `oc get crd <glance CRD>` because in theory creating it with the meta-operator we're not supposed to expose the sub-resources.

**Comment by stuggi** [comment]
> > > @abays @stuggi I'm wondering if we want to enforce `CRDDESC=0` even in `make manifests`
> > 
> > 
> > I'm guessing that removing these descriptions from the bundle and potentially even the CRD manifests will mostly only be noticeable when trying to create the CRs through the OpenShift web console. The fields there would no longer have the descriptions, but I don't think we're supporting GUI creation of resources anyhow. Of course, the `oc get crd <glance CRD>` would not have them either. But I think we need to do this to avoid the problem mentioned in the PR description.
> 
> Yeah, I agree. I would reduce the scope to `oc get crd <glance CRD>` because in theory creating it with the meta-operator we're not supposed to expose the sub-resources.

If possible we should  keep them in the CRD the user is interacting with (ctlplane) that it is not required to check in a doc and not being able to check the meaning of a param from the system.
Also the question is still if the total size ~4MB uncompressed or a single manifest exceeds ~1MB, I guess its the total.

Another point to keep in mind, when we have newer api version in any of the component/s, at least for an intermediate time where we have multiple versions, it will double the size for that component/s.

**Comment by fmount** [comment]
> > > > @abays @stuggi I'm wondering if we want to enforce `CRDDESC=0` even in `make manifests`
> > > 
> > > 
> > > I'm guessing that removing these descriptions from the bundle and potentially even the CRD manifests will mostly only be noticeable when trying to create the CRs through the OpenShift web console. The fields there would no longer have the descriptions, but I don't think we're supporting GUI creation of resources anyhow. Of course, the `oc get crd <glance CRD>` would not have them either. But I think we need to do this to avoid the problem mentioned in the PR description.
> > 
> > 
> > Yeah, I agree. I would reduce the scope to `oc get crd <glance CRD>` because in theory creating it with the meta-operator we're not supposed to expose the sub-resources.
> 
> If possible we should keep them in the CRD the user is interacting with (ctlplane) that it is not required to check in a doc and not being able to check the meaning of a param from the system. Also the question is still if the total size ~4MB uncompressed or a single manifest exceeds ~1MB, I guess its the total.
> 
> Another point to keep in mind, when we have newer api version in any of the component/s, at least for an intermediate time where we have multiple versions, it will double the size for that component/s.

Ack, makes sense. The `request too large` problem I guess is seen only when everything is deployed by the meta-operator [1],
if I deploy the single operator (e.g. through `install_yamls`) I don't see this kind of problem: the question is if we want to address it with a distributed series of patches across the operators or we should find a different strategy at the meta-operator level.

[1] https://paste.opendev.org/show/b9UFDOkEkucoHWDLTKcn/

**Review by Akrog — CHANGES_REQUESTED** [review]
This looks ok since it's in `install` and `deploy` targets as well.

**Review by Akrog — APPROVED** [review]

**Pull request merged**